common/Perf/Azure.Test.Perf/PerfProgram.cs

```diff
@@ -174,6 +178,48 @@ private static async Task Run(Type testType, PerfOptions options)
             PrintAssemblyVersions(testType);
         }
 
+        private static void ConfigureThreadPool(PerfOptions options)
+        {
+            if (options.MinWorkerThreads.HasValue || options.MinCompletionPortThreads.HasValue)
+            {
+                ThreadPool.GetMinThreads(out var minWorkerThreads, out var minCompletionPortThreads);
+                ThreadPool.SetMinThreads(options.MinWorkerThreads ?? minWorkerThreads,
+                    options.MinCompletionPortThreads ?? minCompletionPortThreads);
+            }
+
+            if (options.MaxWorkerThreads.HasValue || options.MaxCompletionPortThreads.HasValue)
+            {
+                ThreadPool.GetMaxThreads(out var maxWorkerThreads, out var maxCompletionPortThreads);
+                ThreadPool.SetMaxThreads(options.MaxWorkerThreads ?? maxWorkerThreads,
+                    options.MaxCompletionPortThreads ?? maxCompletionPortThreads);
+            }
+        }
+
+        private static void PrintEnvironment()
+        {
+            Console.WriteLine("=== Environment ===");
+
+            Console.Write("Configuration: ");
+#if DEBUG
+            Console.WriteLine("Debug");
```

Comment:
We might want to print the same for referenced assemblies.

Reply:
@pakrym: Print what for the referenced assemblies?  We already print the versions:

https://github.com/Azure/azure-sdk-for-net/blob/5398e5a1a0680be47d1abac2c82e821824d46892/common/Perf/Azure.Test.Perf/PerfProgram.cs#L204-L207

Reply:
The configuration (DEBUG/RELEASE).

Reply:
How can I determine this for a reference assembly?

Reply:
`[assembly: AssemblyConfiguration("Debug")]` attribute is set on DEBUG assemblies, at least the ones we produce in azure-sdk-for-net.

Reply:
There is also 
`[assembly: Debuggable(DebuggableAttribute.DebuggingModes.Default | DebuggableAttribute.DebuggingModes.DisableOptimizations | DebuggableAttribute.DebuggingModes.IgnoreSymbolStoreSequencePoints | DebuggableAttribute.DebuggingModes.EnableEditAndContinue)]` but we need to parse it's argument and see if it contains `DebuggableAttribute.DebuggingModes.DisableOptimizations`

Reply:
@pakrym: Added in this commit: https://github.com/Azure/azure-sdk-for-net/pull/20372/commits/078402aa9101d5a7c0c7088df84261cf33b733df.  Open to more feedback on naming, etc.

Reply:
LGTM! thank you